### DISCUSSION TITLE
Execution error on php v7.4 with excluded subcodebase containing php v8+

### DISCUSSION THREAD

**Discussion by 94noni** [PHP-CS-Fixer/PHP-CS-Fixer#6991] [Q&A] [answered]
## Support question

Execution of php-cs-fixer v3.14 with runtime php v7.4 on a codebase that may have some php v8+ code (attributes)

My codebase is like this:

```
// this is mainly a code monolith repository containing bundle/component/etc
  - src
    - Bundle
      - MyBundle (with php v8+ code)
      - MyOtherBundle (with php v7+ code)
    - core (with php v7+ code)
```

Even if I `exclude()` the folder path in my config file, the error outputs: 

```
Files that were not fixed due to errors reported during linting before fixing:                                                                                             
   1) src/Bundle/MyBundle/Twig/CartRequestForm.php  
```

---

- is it even possible ?
- is there a failure in the config file with excluded folder ?

Many thanks,

**Comment by kubawerlos** [comment]
> is there a failure in the config file with excluded folder ?

Definitely, can you share the config?

**Reply by 94noni** [reply]
Will definitly do tomorrow for sure
Thx for stopping by :)

**Comment by 94noni** [comment]
@kubawerlos here it is:

<img width="265" alt="Capture d’écran 2023-05-25 à 07 47 14" src="https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/1358361/13d46b10-06d3-4c52-adc2-1da1003753c9">


```php
<?php

use PhpCsFixer\Config;
use PhpCsFixer\Finder;

$config = new Config();

return $config
    ->setRiskyAllowed(true)
    ->setRules([
        '@PSR1' => true,
        '@PSR2' => true,
        '@Symfony' => true,
        '@Symfony:risky' => true,
        '@DoctrineAnnotation' => true,
        ...
    ])
    ->setFinder(
        Finder::create()
            ->in([
                __DIR__ . '/src',
                __DIR__ . '/tests',
            ])
            ->exclude([
                'src/Company/Bundle/CartRequestBundle',
                // same with this:
                // `__DIR__ . '/src/Company/Bundle/CartRequestBundle',`
            ])
    )
;

```

**Reply by kubawerlos** [reply, marked as answer]
As I suspected, classic mistake with `exclude`:
> // directories passed as argument must be relative to the ones defined with the in() method
$finder->in(__DIR__)->exclude('ruby');

(from https://symfony.com/doc/current/components/finder.html)

**Reply by Wirone** [reply]
In other words: try `Company/Bundle/CartRequestBundle` 😉.

**Reply by 94noni** [reply]
I will try this and report back

I've [read the doc](https://cs.symfony.com/doc/config.html) but found it hard to understand thus my post
Opened [a PR](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6992) to try to improve this a bit, of course feel free to close if you found unrelated

anyway, thank you a lot for your time and answers
